### PR TITLE
Magnetic response

### PR DESCRIPTION
**Pull request by stigrj**
Add driver for magnetic response calculations, including nuclear specific NMR using `H_M_pso` as perturbation. Not sure yet about the accuracy of the latter.

I think I need to fix https://github.com/MRChemSoft/mrcpp/issues/100 before this can be merged, some small-normed response functions become *extremely* large in memory.

### CODE REVIEW

**Comment by ilfreddy** [comment]
Busy with the faculty board all day. I will have a look as soon as I can.

File: src/scf_solver/HelmholtzVector.cpp [hunk 1]
```diff
@@ -62,6 +62,8 @@ HelmholtzVector::HelmholtzVector(double pr, const DoubleVector &l)
  * in the OrbitalVector based on the corresponding lambda_i parameter in the
  * HelmholtzVector. Computes output as: out_i = H_i[phi_i]
  *
+ * NOTE: Helmholtz operator will be apply with _absolute_ precision
```
Comment (ilfreddy) [inline src/scf_solver/HelmholtzVector.cpp]: applied

File: src/scf_solver/HelmholtzVector.cpp [hunk 2]
```diff
@@ -98,6 +100,9 @@ OrbitalVector HelmholtzVector::operator()(OrbitalVector &Phi) const {
  * Specialized version with smaller memory footprint since the full vector V*Phi
  * is never stored, but computed on the fly.
  *
+ * NOTE: Helmholtz operator will be apply with _absolute_ precision
```
Comment (ilfreddy) [inline src/scf_solver/HelmholtzVector.cpp]: applied

File: src/input/mrchem.in [hunk 3]
```diff
@@ -266,21 +269,42 @@ def write_scf_calculation(program_dict, user_dict):
...
+      if all_nucs or k in nuc_vec:
+        prop_dict["nmr_shielding"].append({
+          "nucleus_k": k,
+          "operator": "h_bm_dia",
+          "precision": user_dict["world_prec"],
+          "smoothing": user_dict["world_prec"],
+          "r_O": program_dict["molecule"]["gauge_origin"],
+          "r_K": nuclei[k]["xyz"]
+        })
+  scf_dict["properties"] = prop_dict
```
Comment (ilfreddy) [inline src/input/mrchem.in]: Is the indentation OK here? Looks like i is only two spaces instead of four (it applies to the whole file)

**Comment by stigrj** [comment]
`mrchem.in` really hurts my eyes, but it will be fixed later #247

**Pull request merged**